python_modules/dagster/dagster/_core/definitions/job_definition.py

```diff
@@ -573,6 +574,9 @@ def run_request_for_partition(
                 value means that a run will always be launched per evaluation.
             tags (Optional[Dict[str, str]]): A dictionary of tags (string key-value pairs) to attach
                 to the launched run.
+            run_config (Optional[Mapping[str, Any]]: Configuration for the run. If the job has
```

Comment:
This feels like a foot gun...  they want to do this rather than just create their own RunRequest and add the partition tags?

Granted, the current way of getting the partition tags is awkward, but if we made some small changes, we could enable something like:

```
yield RunRequest(run_key=None, run_config=my_custom_run_config, tags=job.tags_for_partition(my_partition_key)
```

Reply:
When you say it's a footgun, are there situations you have in mind?

The way I've been looking at it is that tags are an implementation detail for assigning a partition to a run. I.e. most users who use partitions likely aren't aware that tags are the mechanism that we use for assigning partitions to runs.  In Dagit, the Launchpad allows you to launch a partitioned run and supply the configuration, without thinking about tags. As a user, I would expect to be able to perform that same operation using code.

Reply:
I was originally thinking that there'd be confusion about what config was used for what when trying to produce a partition run.  But the syntax I proposed doesn't really change that confusion.  Also, I suppose I've been stuck in the mode of equating a job's partition space to its run config, which we might be trying to move away from anyway.

Feel free to ignore.